sievelib/commands.py

```diff
@@ -347,6 +347,19 @@ def __is_valid_value_for_arg(self, arg, value):
             return True
         return value.lower() in arg["values"]
 
+    def __is_valid_type(self, typ, typlist):
+        """ Check if type is valid base on input type list
+            "string" is special because it can be used for stringlist
+
+        :param typ: the type to check
+        :param typlist: the list of type to check
+        :return: True on success, False otherwise
+        """
+        typ_is_str = typ == "string"
+        str_list_in_typlist = "stringlist" in typlist
+
+        return typ not in typlist and not (typ_is_str and str_list_in_typlist)
```

Comment:
```suggestion
        return typ in typlist or (typ_is_str and str_list_in_typlist)
```
Looks easier to understand.